decentraland/unity-explorer
# feat: use entity hash for ABs

## What does this PR change?

There's an issue with ABs and their dependencies. If I have an asset bundle which has external dependecies, and only the external dependecies change, leaving the origin asset bundle hash to remain the same, cache problems arouse. This tries to solve it by

- CDN Cache: Adds the entity hash to the ABs path if the version is greater than 25. 
- Unity Local Cache: Modifies how we use `Hash128`. We are now hashing the date as well, if we didn't, the local asset bundle cache will still return the old ones on a scene redeployment

As a result, each asset is going to be unique for every scene re-deploy

The downside is that assets will not persist during scene re-deployment. If a scene get redeployed, the local cache is invalid.

Requires this ABConverter [PR](https://github.com/decentraland/asset-bundle-converter/pull/156/files) to be merged.

NOTE:

This change affects only alpha. WebGL AB remains untouched, and nothing need to be done in `unity-renderer`

<img width="779" alt="image" src="https://github.com/user-attachments/assets/2db0fd1c-2f7b-4b3e-9ef1-f167f2eebad9">


## How to test the changes?

1. Launch the explorer
2. Try to go to this scene, which should now have the abs. Confirm that you see them
3. Run the explorer normally and check that ABs load on other scenes 

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
